airflow/models/dag.py

```diff
@@ -3706,7 +3706,11 @@ def calculate_dagrun_date_fields(
 
     @provide_session
     def get_dataset_triggered_next_run_info(self, *, session=NEW_SESSION) -> dict[str, int | str] | None:
-        if self.dataset_expression is None:
+        # If an alias is unresolved, we don't have the next_run_info to get
+        if (
+            self.dataset_expression is None
+            and self.timetable_summary != DatasetTriggeredTimetable.UNRESOLVED_ALIAS_SUMMARY
```

Comment:
This part should probably be a bit more robust… maybe add a couple of flags on `DatasetTriggeredTimetable` and use that instead?

Reply:
Or maybe just a flag on Timetable altogether. We have things like `can_be_scheduled`. Maybe a `has_resolved_dataset` would be reasonable?

Reply:
yep, we could do that. `has_resolved_dataset` sounds good

Reply:
I just look into it a bit. It looks like `get_dataset_triggered_next_run_info` is part of `DagModel` instead and I think we probably don't really want to create a new column `has_resolved_dataset`

Reply:
will extracting this condition as a `has_resolved_dataset` property a bit better?

Reply:
The main problem is I don’t think it is a good idea to do this `timetable_summary` check at all.

Reply:
If that's the case, I think we probably need to add an additional boolean column here then?

Reply:
After a second thought, we probably don't need to make things complicated. Just return None if no _dataset_triggered_next_run_info can be retrieved (when there're no dataset to join)